packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -201,40 +217,47 @@ export function makeRumPublicApi(
     }),
 
     setUser: monitor((newUser: User) => {
-      const sanitizedUser = sanitizeUser(newUser)
-      if (sanitizedUser) {
-        user = sanitizedUser
-      } else {
+      if (typeof newUser !== 'object' || !newUser) {
         display.error('Unsupported user:', newUser)
+      } else {
+        userContextManager.setContext(sanitizeUser(newUser as Context))
       }
     }),
 
+    getUser: monitor(userContextManager.getContext),
+
+    setUserProperty: monitor(userContextManager.setContextProperty),
+
+    removeUserProperty: monitor(userContextManager.removeContextProperty),
+
+    clearUser: monitor(userContextManager.clearContext),
+
     removeUser: monitor(() => {
-      user = {}
+      userContextManager.clearContext()
     }),
 
+    addUserAttribute: monitor(userContextManager.setContextProperty),
+
+    removeUserAttribute: monitor(userContextManager.removeContextProperty),
+
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏those should be removed